### ISSUE TITLE
Using font size greater than 16 cause docs.rs to render oddly

### ISSUE THREAD

**Issue by iamsauravsharma** [rust-lang/docs.rs#1607]
Changing default font size from 16 to 17 on Firefox makes docs.rs page to render oddly. 

### Default font size: 17
![Screenshot_20220117_160326](https://user-images.githubusercontent.com/38726015/149751591-ad86464d-0bd2-409e-9ec1-e98cc46b0661.png)

### Default font size: 16
![Screenshot_20220117_160350](https://user-images.githubusercontent.com/38726015/149751666-0c967495-003f-40f3-8da1-742b4cdbeec3.png)

**Comment by kinwolfqc** [comment]
Same behavior in Chrome too if you change the Font Size setting to anything higher then "medium"

**Comment by jsha** [comment]
That might be related to https://github.com/rust-lang/rust/pull/92448, which changed our font sizes from `px` (absolute) to `rem` (relative to the user's font size). I don't see how that could produce the symptoms you're seeing, but it's related to font sizes.

Do you see this on all crates, or just recently built crates?

**Comment by kinwolfqc** [comment]
@jsha Indeed, this seems to only happens on newer crates.  Even older version of a crate in problem will display properly, but the most recent one has the problem(ex: https://docs.rs/axum/0.2.4/axum/index.html is fine, but https://docs.rs/axum/latest/axum/index.html has the problem.

**Comment by jsha** [comment]
A little quick investigation shows this is because the `<main>` element is in a flexbox alongside the sidebar, and the browser is choosing to shrink it from its default size (but seemingly only when the font size is bigger than default?).

Setting `flex-shrink: 0` on `<main>` seems to fix, but I'm not sure that's the correct solution.

**Comment by jsha** [comment]
Also, unchecking the `main { min-width: 0 }` rule (or setting it to `auto` seems to do the trick. https://developer.mozilla.org/en-US/docs/Web/CSS/min-width

**Comment by jsha** [comment]
@cynecx in your [PR fixing the sidebar to use flexbox](https://github.com/rust-lang/rust/pull/89385/files#diff-67c3587d759aa9578993149e68eb8d3a9de6b28e5c0b85091b0a71a07f441795R244-R249) (thanks for that!), you had:

```
main {
	position: relative;
	flex-grow: 1;
	padding: 10px 15px 40px 45px;
	min-width: 0;
}
```

Do you happen to remember why you set min-width: 0?

**Comment by cynecx** [comment]
> Setting flex-shrink: 0 on <main> seems to fix, but I'm not sure that's the correct solution.

Yeah, I don't think that's the proper fix.

> Do you happen to remember why you set min-width: 0?

I don't quite remember but I think it was because flex child elements have `min-width` defaulted to auto and it might overflow the parent.

But I don't think that's the proper fix here.

If we look at the flex container:

![image](https://user-images.githubusercontent.com/5961244/150070559-a3725b13-efeb-4dc3-a855-b2163549222c.png)

We can see that the flexbox container isn't utilizing the whole width (because it hasn't been set!). So perhaps the proper fix would be to make the flex container `width: 100%`? (I've tested it locally and it seems to work just fine)
